mapof.go

```diff
@@ -56,22 +56,31 @@ type IntegerConstraint interface {
 
 // NewIntegerMapOf creates a new MapOf instance with integer typed keys.
 func NewIntegerMapOf[K IntegerConstraint, V any]() *MapOf[K, V] {
-	return NewTypedMapOf[K, V](func(k K) uint64 {
-		return uint64(k)
-	})
+	return NewSeededTypedMapOf[K, V](hash64[K])
 }
 
 // NewTypedMapOf creates a new MapOf instance with arbitrarily typed keys.
 // Keys are hashed to uint64 using the hasher function. Note that StrHash64
 // function might be handy when writing the hasher function for structs with
 // string fields.
+//
+// Deprecated: Use NewSeededTypedMapOf.
 func NewTypedMapOf[K comparable, V any](hasher func(K) uint64) *MapOf[K, V] {
+	return NewSeededTypedMapOf[K, V](func(seed maphash.Seed, k K) uint64 {
+		// Mitigate poor quality hash functions.
+		return hash64(seed, hasher(k))
+	})
+}
+
+// NewSeededTypedMapOf creates a new MapOf instance with arbitrarily typed keys.
+// Keys are hashed to uint64 using the hasher function. It is strongly adviced
+// to use the hash/maphash package to implement hasher. See the Example for how
+// to do that.
+func NewSeededTypedMapOf[K comparable, V any](hasher func(maphash.Seed, K) uint64) *MapOf[K, V] {
```

Comment:
I'd better change the signature of `NewTypedMapOf`. Having one more function doesn't make sense. Incompatible change is fine since there are not so many users of `NewTypedMapOf` and I'm going to release v2.0 with these changes.